File: app/src/main/java/io/palaima/calendar/data/MyEvent.java
package io.palaima.calendar.data;

import android.graphics.Canvas;
import android.graphics.RectF;
import android.support.annotation.NonNull;

import java.util.Date;

import io.palaima.eventscalendar.data.CalendarEvent;

public class MyEvent extends CalendarEvent {

    public MyEvent(@NonNull Date startTime, @NonNull Date endTime, long categoryId) {
        super(startTime, endTime, categoryId);
    }

    @Override protected void onDraw(@NonNull Canvas canvas, @NonNull RectF eventRect) {

    }

    public static MyEvent from(EventEntity eventEntity) {
        return new MyEvent(eventEntity.startDate().getTime(), eventEntity.endDate().getTime(), eventEntity.categoryId());
    }
}

File: app/src/main/java/io/palaima/calendar/data/MyEvent.kt
package io.palaima.calendar.data

import android.graphics.Canvas
import android.graphics.RectF
import io.palaima.eventscalendar.data.CalendarEvent
import java.util.*

class MyEvent(startTime: Date, endTime: Date, categoryId: Long) : CalendarEvent<String>("hello world", startTime, endTime, categoryId) {

    override fun onDraw(canvas: Canvas, eventRect: RectF) {

    }

    companion object {

        fun from(eventEntity: EventEntity): MyEvent {
            return MyEvent(Date(eventEntity.startDate()), Date(eventEntity.endDate()), eventEntity.categoryId())
        }
    }
}
